src/Pages/Order.tsx

```diff
@@ -1,165 +1,230 @@
+    resolver: zodResolver(FormSchema),
+    defaultValues: { senderName: '', message: '' },
+    mode: 'onSubmit',
+  });
+
+  const handleSelectCard = (card: typeof selectedCard) => {
+    selectCard(card!);
+    setValue('message', card!.defaultTextMessage);
+    hasUserEditedMessage.current = false;
+  };
+
+  const onMessageChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setValue('message', e.target.value);
+  };
 
   const hasUserEditedMessage = useRef(false);
 
-  const handleMessageChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
-    hasUserEditedMessage.current = true;
-    message.onChange(e);
+  const [isReceiverModalOpen, setIsReceiverModalOpen] = useState(false);
+  const [receivers, setReceivers] = useState<Receiver[]>([]);
+  const [editingReceivers, setEditingReceivers] = useState<Receiver[] | null>(
+    null
+  );
+  const [receiverError, setReceiverError] = useState(false);
+
+  const handleOpenReceiverModal = () => {
+    setEditingReceivers(receivers.length > 0 ? receivers : null);
+    setIsReceiverModalOpen(true);
+  };
+
+  const handleModalAddOrEdit = (newReceivers: Receiver[]) => {
+    setReceivers(newReceivers);
+    setEditingReceivers(null);
+    setIsReceiverModalOpen(false);
+  };
+
+  const handleModalClose = () => {
+    setIsReceiverModalOpen(false);
+    setEditingReceivers(null);
   };
 
   useEffect(() => {
-    if (selectedCard?.defaultTextMessage && !hasUserEditedMessage.current && message.value === '') {
-      message.onChange({
-        target: { value: selectedCard.defaultTextMessage },
-      } as React.ChangeEvent<HTMLTextAreaElement>);
+    if (
+      selectedCard?.defaultTextMessage &&
+      !hasUserEditedMessage.current &&
+      getValues('message') === ''
+    ) {
+      setValue('message', selectedCard.defaultTextMessage);
     }
-  }, [selectedCard, message]);
+  }, [selectedCard, getValues, setValue]);
 
   const navigate = useNavigate();
 
   if (!item) return <p>상품 정보를 찾을 수 없습니다.</p>;
 
-  const handleOrderSubmit = () => {
-    const isMessageVaild = message.validate();
-    const isSenderNameValid = senderName.validate();
-    const isReceiverNameValid = receiverName.validate();
-    const isPhoneValid = receiverPhoneNumber.validate();
-    const isItemCountValid = itemCount.validate();
-
-    const valid =
-      isMessageVaild &&
-      isSenderNameValid &&
-      isReceiverNameValid &&
-      isPhoneValid &&
-      isItemCountValid;
-    if (valid) {
-      alert(
-        `주문이 완료되었습니다.\n상품명: ${item?.name}\n구매 수량: ${itemCount.value}\n발신자 이름: ${senderName.value}\n메시지: ${message.value}`
-      );
-      navigate('/');
+  const totalCount = receivers.reduce(
+    (acc, curr) => acc + (curr.itemCount ?? 0),
+    0
+  );
+  const totalPrice = totalCount * (item?.price.sellingPrice ?? 0);
+
+  const handleOrderSubmit = (data: FormFields) => {
+    if (receivers.length < 1) {
+      setReceiverError(true);
+      return;
     }
+    setReceiverError(false);
```

Comment:
```suggestion
  setReceiverError(receivers.length < 1)
```

요렇게 하면 단순화할 수 있을 거 같네요!